Handbrake: problems updating to 1.3.0

#### Description

As a maintainer of `handbrake` I open this issue before some user would open it or would try to solve issue.

Handbrake is a pretty messy build project. There are strange issues updating to `1.3.0`, not counting some moments they not put into documentation. `nix` build system is not designed for such messy drag-everything-home and "do not respect package managers" build project. For example, now project requires and checks for the `autoconf, autotools, cmake, meson, ninja` all at the same time, but project builds through `autoconf, autotools` and everything other is required as dependency because they blob-in, copy and patch tens of other multimedia upstreams and use those only their-own patched versions for their builds. When I show them code they freak out that we are using dependency management and upstream projects for work with their project. And `handbrake` upstream does not expect to be working in `nix` clean conditions and does not care for it, they only expect everyone to be using their Flatpack, regarding my work on Nix - phrases "this is only another package manager after 100 of other ones, there are already Flatpacks, Snaps and Docker..." phrases was tossed to me (ironically, I am silent with them that they tell this to the person that much before Flatpacks, Snaps was first heard of, was trying to improve Nix adoption by work on improving Nix installation on all platforms, but that work ironically was not accepted, so that my agenda was not met), that is what was drawn by me from several previous discussions. Some main upstream dev is pretty toxic and hard to keep any healthy dialog with. I am very careful in my words, especially with them. I would try to build a dialog, but every time I talk with them, some of their dev escalates the situation to unreasonable proportions and starts to make technical threats to pull off and ruin any support and more-over veto and block Nix on their side, it is when I came to help them, do work and have a dialog on a technical question.

So don't hold your breath.

**Syndrome**
```
  : /nix/store/ngg9lxh4xpnwdidv9sc98dg83vypm78k-binutils-2.31.1/bin/ld: /build/HandBrake-1.3.0/build//libhb/libhandbrake.a(ports.o): undefined reference to symbol 'dlclose@@GLIBC_2.2.5'
  : /nix/store/ngg9lxh4xpnwdidv9sc98dg83vypm78k-binutils-2.31.1/bin/ld: /nix/store/qb6k4hp7gk331x9fydw0w7qj4dv09bwz-glibc-2.27/lib/libdl.so.2: error adding symbols: DSO missing from command line
  : collect2: error: ld returned 1 exit status
  : make[3]: *** [Makefile:623: ghb] Error 1
  : make[3]: Leaving directory '/build/HandBrake-1.3.0/build/gtk/src'
  : make[2]: *** [Makefile:445: all-recursive] Error 1
  : make[2]: Leaving directory '/build/HandBrake-1.3.0/build/gtk'
  : make[1]: *** [Makefile:377: all] Error 2
  : make[1]: Leaving directory '/build/HandBrake-1.3.0/build/gtk'
  : make: *** [../gtk/module.rules:30: gtk.build] Error 2
-------------------------------------------------------------------------------
time end: Thu Nov 14 14:55:37 2019
duration: 50 seconds (50.64s)
result: FAILURE (code 2)
-------------------------------------------------------------------------------
Build is finished!
You may now cd into ./build and examine the output.
building
build flags: -j1 -l8
ninja: error: loading 'build.ninja': No such file or directory
builder for '/nix/store/n5z3ragd8as8hwy8maj20bdj2yp3ik7h-handbrake-1.3.0.drv' failed with exit code 1
error: build of '/nix/store/n5z3ragd8as8hwy8maj20bdj2yp3ik7h-handbrake-1.3.0.drv' failed
```

So far extending with which seems asked `ell` lib and extending build config flags with "LIBS='-ldl'" did not help.

```
  : make[3]: *** [Makefile:623: ghb] Error 1
  : make[3]: Leaving directory '/build/HandBrake-1.3.0/build/gtk/src'
  : make[2]: *** [Makefile:445: all-recursive] Error 1
  : make[2]: Leaving directory '/build/HandBrake-1.3.0/build/gtk'
  : make[1]: *** [Makefile:377: all] Error 2
  : make[1]: Leaving directory '/build/HandBrake-1.3.0/build/gtk'
  : make: *** [../gtk/module.rules:30: gtk.build] Error 2
```

Wait, is it trying to build GTK?

Weirdly, it does not seem to require any special treatment in Arch.

https://git.archlinux.org/svntogit/community.git/tree/trunk/PKGBUILD?h=packages/handbrake

```
ninja: error: loading 'build.ninja': No such file or directory
```

You will want to disable ninja setup hook using

```
dontUseNinjaBuild = true;
dontUseNinjaInstall = true;
dontUseNinjaCheck = true;
```

--- a/pkgs/applications/video/handbrake/default.nix
+++ b/pkgs/applications/video/handbrake/default.nix
@@ -9,7 +9,7 @@
 
 { stdenv, lib, fetchurl, fetchpatch,
   # Main build tools
-  python2, pkgconfig, autoconf, automake, cmake, nasm, libtool, m4, lzma,
+  python2, pkgconfig, autoconf, automake, libtool, m4, lzma,
   numactl,
   # Processing, video codecs, containers
   ffmpeg-full, nv-codec-headers, libogg, x264, x265, libvpx, libtheora, dav1d,
@@ -57,7 +57,7 @@ stdenv.mkDerivation rec {
   };
 
   nativeBuildInputs = [
-    python2 pkgconfig autoconf automake cmake nasm libtool m4
+    python2 pkgconfig autoconf automake libtool m4
   ] ++ lib.optionals useGtk [ intltool wrapGAppsHook ];
 
   buildInputs = [
@@ -75,13 +75,6 @@ stdenv.mkDerivation rec {
   # look at ./make/configure.py search "enable_nvenc"
     ++ lib.optional stdenv.isLinux nv-codec-headers;
 
-  # NOTE: 2018-12-25: v1.2.0 now requires cmake dep
-  # (default distribution bundles&builds 3rd party libs),
-  # don't trigger cmake build
-  dontUseCmakeConfigure = true;
-  # cp: cannot create regular file './internal_defaults.json': File exists
-  enableParallelBuilding = false;
-
   preConfigure = ''
     patchShebangs scripts
 
@@ -93,7 +86,9 @@ stdenv.mkDerivation rec {
     # Force using nixpkgs dependencies
     sed -i '/MODULES += contrib/d' make/include/main.defs
     sed -i '/PKG_CONFIG_PATH=/d' gtk/module.rules
-    sed -i 's/^[[:space:]]*\(meson\|ninja\)[[:space:]]*= ToolProbe.*$//g' make/configure.py
+    sed -e 's/^[[:space:]]*\(meson\|ninja\|nasm\)[[:space:]]*= ToolProbe.*$//g' \
+        -e '/    ## Additional library and tool checks/,/    ## MinGW specific library and tool checks/d' \
+        -i make/configure.py
   '';
 
   configureFlags = [

--- a/pkgs/applications/video/handbrake/default.nix
+++ b/pkgs/applications/video/handbrake/default.nix
@@ -10,8 +10,9 @@
 { stdenv, lib, fetchurl, fetchpatch,
   # Main build tools
   python2, pkgconfig, autoconf, automake, cmake, nasm, libtool, m4, lzma,
+  numactl,
   # Processing, video codecs, containers
-  ffmpeg-full, nv-codec-headers, libogg, x264, x265, libvpx, libtheora,
+  ffmpeg-full, nv-codec-headers, libogg, x264, x265, libvpx, libtheora, dav1d,
   # Codecs, audio
   libopus, lame, libvorbis, a52dec, speex, libsamplerate,
   # Text processing
@@ -48,11 +49,11 @@ assert stdenv.isDarwin -> AudioToolbox != null && Foundation != null
 
 stdenv.mkDerivation rec {
   pname = "handbrake";
-  version = "1.2.2";
+  version = "1.3.0";
 
   src = fetchurl {
     url = ''https://download2.handbrake.fr/${version}/HandBrake-${version}-source.tar.bz2'';
-    sha256 = "0k2yaqy7zi06k8mkp9az2mn9dlgj3a1339vacakfh2nn2zsics6z";
+    sha256 = "15hxncswmaj62hb40fxixsa6d519zb712z9xbdq979q4rasjxa59";
   };
 
   nativeBuildInputs = [
@@ -60,10 +61,10 @@ stdenv.mkDerivation rec {
   ] ++ lib.optionals useGtk [ intltool wrapGAppsHook ];
 
   buildInputs = [
-    ffmpeg-full libogg libtheora x264 x265 libvpx
+    ffmpeg-full libogg libtheora x264 x265 libvpx dav1d
     libopus lame libvorbis a52dec speex libsamplerate
     libiconv fribidi fontconfig freetype libass jansson libxml2 harfbuzz
-    libdvdread libdvdnav libdvdcss libbluray lzma
+    libdvdread libdvdnav libdvdcss libbluray lzma numactl
   ] ++ lib.optionals useGtk [
     glib gtk3 libappindicator-gtk3 libnotify
     gst_all_1.gstreamer gst_all_1.gst-plugins-base dbus-glib udev
@@ -81,13 +82,6 @@ stdenv.mkDerivation rec {
   # cp: cannot create regular file './internal_defaults.json': File exists
   enableParallelBuilding = false;
 
-  # The samplerate patch should be removed when HandBrake 1.3.0 is released
-  patches = [(fetchpatch {
-    name = "set-ffmpeg-samplerate.patch";
-    url = "https://patch-diff.githubusercontent.com/raw/HandBrake/HandBrake/pull/2126.patch";
-    sha256 = "00lds9h27cvyr53qpvv8gbv01hfxdxd8gphxcwbwg8akqrvk9gbf";
-  })];
-
   preConfigure = ''
     patchShebangs scripts
 
@@ -99,6 +93,7 @@ stdenv.mkDerivation rec {
     # Force using nixpkgs dependencies
     sed -i '/MODULES += contrib/d' make/include/main.defs
     sed -i '/PKG_CONFIG_PATH=/d' gtk/module.rules
+    sed -i 's/^[[:space:]]*\(meson\|ninja\)[[:space:]]*= ToolProbe.*$//g' make/configure.py
   '';
 
   configureFlags = [
@@ -112,6 +107,8 @@ stdenv.mkDerivation rec {
   # NOTE: 2018-12-27: Check NixOS HandBrake test if changing
   NIX_LDFLAGS = [
     "-lx265"
+    # NOTE: The -ldl flag was fixed upstream for a release after 1.3.0    
+    "-ldl"
   ];
 
   preBuild = ''

